cliconfig/config.go

```diff
@@ -29,20 +31,15 @@ var (
 	configDir = os.Getenv("DOCKER_CONFIG")
 )
 
-func getDefaultConfigDir(confFile string) string {
-	confDir := filepath.Join(homedir.Get(), confFile)
-	// if the directory doesn't exist, maybe we called docker with sudo
-	if _, err := os.Stat(configDir); err != nil {
-		if os.IsNotExist(err) {
-			return filepath.Join(homedir.GetWithSudoUser(), confFile)
-		}
-	}
-	return confDir
-}
-
 func init() {
 	if configDir == "" {
-		configDir = getDefaultConfigDir(".docker")
+		configDir = filepath.Join(homedir.Get(), defaultConfigDir)
+		if _, err := os.Stat(configDir); err != nil {
+			// if the directory doesn't exist, maybe we called docker with sudo
```

Comment:
I would prefer that solution because then the user explicitly asked for us to touch *their* config file.
Having `sudo` run as root but touch someone else's files is confusing.